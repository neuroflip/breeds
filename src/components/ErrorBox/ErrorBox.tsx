
import ErrorBoxDiv from "./styled/ErrorBoxDiv"

type ErrorBoxProps = {
  errorMessage: string
}

const ErrorBox = ({ errorMessage }: ErrorBoxProps) => {
  return (<ErrorBoxDiv aria-label="Error content" role="alert" aria-live='polite'>
    <p>{ errorMessage }</p>
    <p>Please, retry again later</p>
  </ErrorBoxDiv>)
}

export default ErrorBox