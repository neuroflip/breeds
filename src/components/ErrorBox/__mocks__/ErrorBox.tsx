const ErrorBox = () => {
  return <>ErrorBox</>
}

export default ErrorBox