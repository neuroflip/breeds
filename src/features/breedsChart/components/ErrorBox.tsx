
type ErrorBoxProps = { 
  errorMessage: string
} 

const ErrorBox = ({ errorMessage }: ErrorBoxProps) => {
  return (<>
    <p>
      { errorMessage }
    </p>
    <p>Please, retry again later</p>
  </>)
}

export default ErrorBox