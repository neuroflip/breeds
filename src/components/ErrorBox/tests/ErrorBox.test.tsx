import { describe, expect, test } from 'vitest';
import { render, screen } from '@testing-library/react';
import ErrorBox from '../ErrorBox';

const inputErrorMessage = "Something went wrong"
const defaultErrorMessage = "Please, retry again later"

describe('Error Box', () => {
  test('initial render', () => {
    render(<ErrorBox errorMessage={ inputErrorMessage } />);

    const errorBoxElement = screen.getByRole('alert');
    const inputErrorMessageElement = screen.getByText(inputErrorMessage);
    const standardErrorMessageElement = screen.getByText(defaultErrorMessage);

    expect(errorBoxElement).toBeInTheDocument();
    expect(inputErrorMessageElement).toBeInTheDocument();
    expect(standardErrorMessageElement).toBeInTheDocument();
    expect(errorBoxElement.className).toBe('errorBox');
    expect(errorBoxElement.getAttribute('role')).toBe('alert');
  });
});