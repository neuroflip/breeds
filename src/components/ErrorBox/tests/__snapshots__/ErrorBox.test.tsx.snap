// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Error Box > initial render 1`] = `
<div>
  <div
    aria-label="Error content"
    aria-live="polite"
    class="sc-VHjGu eslFRO"
    role="alert"
  >
    <p>
      Something went wrong
    </p>
    <p>
      Please, retry again later
    </p>
  </div>
</div>
`;
